src/StreamJsonRpc/JsonRpc.cs

```diff
@@ -1347,6 +1268,138 @@ public void Dispose()
             GC.SuppressFinalize(this);
         }
 
+        /// <summary>
+        /// Creates a marshallable proxy for a given object that may be sent over RPC such that the receiving side can invoke methods on the given object.
+        /// </summary>
+        /// <typeparam name="T">
+        /// The interface type implemented by the <paramref name="marshaledObject"/> that defines the members to expose over RPC.
+        /// </typeparam>
+        /// <param name="marshaledObject">The object to be exposed over RPC.</param>
+        /// <param name="options"><inheritdoc cref="RpcMarshaledContext{T}(T, JsonRpcTargetOptions)" path="/param[@name='options']"/></param>
+        /// <returns>A lifetime controlling wrapper around a new proxy value.</returns>
+        /// <remarks>
+        /// <para>
+        /// Use <see cref="MarshalLimitedArgument{T}(T)"/> for a simpler lifetime model when the object should only be marshaled within the scope of a single RPC call.
+        /// </para>
+        /// </remarks>
+        internal static IRpcMarshaledContext<T> MarshalWithControlledLifetime<T>(T marshaledObject, JsonRpcTargetOptions options)
```

Comment:
If you think it would help, I'm open to it. I tend to keep private nested classes in the main file but this _is_ getting to be a rather large file. I would do it as a separate PR though.